backend/services/office_hours/event.py

```diff
@@ -0,0 +1,679 @@
+            if current_date.strftime("%A").lower() in weekday_names:
+                dates_in_range.append(current_date)
+            # Move to the next day
+            current_date += timedelta(days=1)
+
+        return dates_in_range
+
+    def _transformDate(self, orginal_date: datetime, new_event_date: date) -> datetime:
+
+        # Extract time components from the original datetime
+        hour = orginal_date.hour
+        minute = orginal_date.minute
+
+        # Combine the new date with the time components
+        new_datetime = datetime.combine(new_event_date, datetime.min.time())
+
+        # Set the time components to the new datetime
+        new_datetime = new_datetime.replace(hour=hour, minute=minute)
+
+        return new_datetime
+
+    def update(
+        self, subject: User, oh_event_delta: OfficeHoursEventPartial
+    ) -> OfficeHoursEvent:
+        """Updates an office hours event.
+
+        Args:
+            subject (User): a valid User model representing the currently logged in User
+            oh_event_delta (OfficeHoursEventPartial): OfficeHoursEventPartial delta to update in the table
+
+        Returns:
+            OfficeHoursEvent: Updated object in table
+
+        Raises:
+            PermissionError: If the user attempting to update the event is a student.
+
+        """
+        oh_event_entity = self._session.get(OfficeHoursEventEntity, oh_event_delta.id)
+
+        if oh_event_entity is None:
+            raise ResourceNotFoundException(
+                f"Could not find Office Hours event with id={oh_event_delta.id}"
+            )
+
+        section_member_entity = self._check_user_section_membership(
+            subject.id, oh_event_entity.office_hours_section_id
+        )
+
+        if section_member_entity.member_role == RosterRole.STUDENT:
+            raise PermissionError(
+                "User is a Student. Do not have permission to update an event."
+            )
+
+        # Update Delta Fields
+        if oh_event_delta.event_date is not None:
+            oh_event_entity.date = oh_event_delta.event_date
+
+        if oh_event_delta.start_time is not None:
+            oh_event_entity.start_time = oh_event_delta.start_time
+
+        if oh_event_delta.end_time is not None:
+            oh_event_entity.end_time = oh_event_delta.end_time
+
+        if oh_event_delta.location_description is not None:
+            oh_event_entity.location_description = oh_event_delta.location_description
+
+        if oh_event_delta.description is not None:
+            oh_event_entity.description = oh_event_delta.description
+
+        if oh_event_delta.type is not None:
+            oh_event_entity.type = oh_event_delta.type
+
+        if oh_event_delta.room is not None:
+            oh_event_entity.room_id = oh_event_delta.room.id
+
+        if oh_event_delta.oh_section is not None:
+            raise Exception("Updating Office Hours Event OH Section is Not Allowed.")
```

Comment:
Since we are using an office hours section partial, I was initially anticipating that the partial would only pass in the updated fields -- so that's why checking none would indicate an update. However, we may end up just passing the whole object on frontend, so I agree the check may need to be that